BridgeAppSDK/SBAPermissionsManager.swift

```diff
@@ -33,97 +33,73 @@
-        case "healthKit":
-            self.init(rawValue:SBAPermissionsType.healthKit.rawValue)
-        case "location":
-            self.init(rawValue:SBAPermissionsType.location.rawValue)
-        case "localNotifications":
-            self.init(rawValue:SBAPermissionsType.localNotifications.rawValue)
-        case "microphone":
-            self.init(rawValue:SBAPermissionsType.microphone.rawValue)
-        case "photoLibrary":
-            self.init(rawValue:SBAPermissionsType.photoLibrary.rawValue)
-        default:
-            let intValue = UInt(key) ?? 0
-            self.init(rawValue: intValue)
-        }
-    }
-    
-    var items: [UInt] {
-        var items: [UInt] = []
-        for ii in UInt(0)...UInt(16) {
-            let rawValue = 1 << ii
-            let member = SBAPermissionsType(rawValue: rawValue)
-            if self.contains(member) {
-                items.append(rawValue)
-            }
-        }
-        return items
-    }
-}
-
 extension SBAPermissionsManager {
     
     public static var shared: SBAPermissionsManager {
         return __shared()
     }
     
-    public func requestPermissions(_ permissions: SBAPermissionsType, alertPresenter: SBAAlertPresenter?, completion: ((Bool) -> Void)?) {
-        
+    public func requestPermissions(for permissions: [SBAPermissionObjectType], alertPresenter: SBAAlertPresenter?, completion: ((Bool) -> Void)?) {
+    
         // Exit early if there are no permissions
-        let items = permissions.items
-        guard items.count > 0 else {
+        guard permissions.count > 0 else {
             completion?(true)
             return
         }
-        
-        // Use a dispatch group to iterate through all the permissions and accept them as
-        // a batch, showing an alert for each if not authorized.
-        let dispatchGroup = DispatchGroup()
-        var allGranted = true
-        
-        for item in items {
-            let permission = SBAPermissionsType(rawValue: item)
-            if !self.isPermissionsGranted(for: permission) {
-                dispatchGroup.enter()
-                self.requestPermission(for: permission, withCompletion: { [weak alertPresenter] (success, error) in
-                    allGranted = allGranted && success
-                    if !success, let presenter = alertPresenter {
+
+        DispatchQueue.main.async(execute: {
+            
+            // Use an enumerator to recursively step thorough each permission and request permission
+            // for that type.
+            var allGranted = true
+            let eumerator = (permissions as NSArray).objectEnumerator()
+            
+            func recursiveRequest() {
+                guard let permission = eumerator.nextObject() as? SBAPermissionObjectType else {
+                    completion?(allGranted)
+                    return
+                }
+
+                if self.isPermissionGranted(for: permission) {
+                    recursiveRequest()
```

Comment:
Isn't this just going to iterate through the permissions list? Why recursively? There's no parameter to the recursive function for it to do recursive-descent on, and there doesn't seem to be any variable carried through to recursive scopes other than where it is in the (apparently linear) permissions enumerator which is defined outside the function, so why not do this in a for loop using continue statements instead of recursively calling itself? Am I missing something?